src/pip/_internal/req/req_install.py

```diff
@@ -265,6 +268,7 @@ def populate_link(self, finder, upgrade, require_hashes):
                 supported_tags=supported_tags,
             )
             if old_link != self.link:
+                self.from_wheel_cache = True
```

Comment:
I think `self.from_wheel_cache` would be better as a property, that returns a boolean indicating whether the current link points to a path within `self.wheel_cache.cache_dir`.

That'd decouple the value from the specific code-flow and instead depend directly on the "source of truth". This would be good to have, since `populate_link` would no longer be used by the new pip resolver that's being worked on (https://pyfound.blogspot.com/2020/03/new-pip-resolver-to-roll-out-this-year.html).